google-cloud-spanner/acceptance/spanner/client/transaction_test.rb

```diff
@@ -126,6 +128,83 @@
       assert_accounts_equal expected, actual
     end
   end
+  focus
+  it "supports tx isolation with read and update" do
+    results = db.read "accounts", [:reputation], keys: 1, limit: 1
+    original_val = results.rows.first[:reputation]
+    begin
+      latch = Concurrent::CountDownLatch.new 1
+      thr_1 = Thread.new do
+        db.transaction do |tx|
+          tx_results = tx.read "accounts", [:reputation], keys: 1, limit: 1
+          puts "read 1"
+          tx_val = tx_results.rows.first[:reputation]
+          new_val = tx_val + 1
+          tx.update "accounts", [{ account_id: 1, reputation: new_val }]
+          puts "write 1"
+          latch.wait
```

Comment:
In Java, the buffer method just buffers the mutations in memory and does not make the commit call. Commit is made automatically by the library when the callable returns. So it is waiting before the commit rpc is made. But here I think the update call also makes the rpc, so you are waiting after the rpc. @blowmage is correct in that you should wait before the commit. Then only you can guarantee that both the threads have read the value before they try to commit.